feature/experimental/p4rt/ate_tests/lldp_packetin_test/lldp_packetin_test.go

```diff
@@ -0,0 +1,492 @@
+								t.Fatalf("Packet in PacketIn message is not matching wanted packet.")
+							}
+						}
+
+						metaData := packet.Pkt.GetMetadata()
+						for _, data := range metaData {
+							if data.GetMetadataId() == METADATA_INGRESS_PORT {
+								if string(data.GetValue()) != args.packetIO.GetIngressPort() {
+									t.Fatalf("Ingress Port Id is not matching expectation.")
+								}
+							}
+							if data.GetMetadataId() == METADATA_EGRESS_PORT {
+								found := false
+								for _, portData := range args.packetIO.GetEgressPort() {
+									if string(data.GetValue()) == portData {
+										found = true
+									}
+								}
+								if !found {
+									t.Fatalf("Egress Port Id is not matching expectation.")
+								}
+
+							}
+						}
+					}
+				}
+			}
+		})
+	}
+}
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// sortPorts sorts the ports by the testbed port ID.
+func sortPorts(ports []*ondatra.Port) []*ondatra.Port {
+	sort.Slice(ports, func(i, j int) bool {
+		idi, idj := ports[i].ID(), ports[j].ID()
+		li, lj := len(idi), len(idj)
+		if li == lj {
+			return idi < idj
+		}
+		return li < lj // "port2" < "port10"
+	})
+	return ports
+}
+
+// configInterfaceDUT configures the interface with the Addrs.
+func configInterfaceDUT(i *telemetry.Interface, a *attrs.Attributes) *telemetry.Interface {
+	i.Description = ygot.String(a.Desc)
+	i.Type = telemetry.IETFInterfaces_InterfaceType_ethernetCsmacd
+	if *deviations.InterfaceEnabled {
+		i.Enabled = ygot.Bool(true)
+	}
+
+	s := i.GetOrCreateSubinterface(0)
+	s4 := s.GetOrCreateIpv4()
+	if *deviations.InterfaceEnabled {
+		s4.Enabled = ygot.Bool(true)
+	}
+	s4a := s4.GetOrCreateAddress(a.IPv4)
+	s4a.PrefixLength = ygot.Uint8(ipv4PrefixLen)
+
+	return i
+}
+
+// configureDUT configures port1 and port2 on the DUT.
+func configureDUT(t *testing.T, dut *ondatra.DUTDevice) {
+	d := dut.Config()
+
+	p1 := dut.Port(t, "port1")
+	i1 := &telemetry.Interface{Name: ygot.String(p1.Name())}
+	d.Interface(p1.Name()).Replace(t, configInterfaceDUT(i1, &dutPort1))
+
+	p2 := dut.Port(t, "port2")
+	i2 := &telemetry.Interface{Name: ygot.String(p2.Name())}
+	d.Interface(p2.Name()).Replace(t, configInterfaceDUT(i2, &dutPort2))
+}
+
+// configureATE configures port1 and port2 on the ATE.
+func configureATE(t *testing.T, ate *ondatra.ATEDevice) *ondatra.ATETopology {
+	top := ate.Topology().New()
+
+	p1 := ate.Port(t, "port1")
+	i1 := top.AddInterface(atePort1.Name).WithPort(p1)
+	i1.IPv4().
+		WithAddress(atePort1.IPv4CIDR()).
+		WithDefaultGateway(dutPort1.IPv4)
+
+	p2 := ate.Port(t, "port2")
+	i2 := top.AddInterface(atePort2.Name).WithPort(p2)
+	i2.IPv4().
+		WithAddress(atePort2.IPv4CIDR()).
+		WithDefaultGateway(dutPort2.IPv4)
+
+	return top
+}
+
+// configureDeviceId configures p4rt device-id on the DUT.
+func configureDeviceId(ctx context.Context, t *testing.T, dut *ondatra.DUTDevice) {
```

Comment:
func configureDeviceId should be configureDeviceID

Words in names that are initialisms or acronyms (e.g., URL and NATO) should have the same case. URL should appear as URL or url (as in urlPony, or URLPony), never as Url. This also applies to ID when it is short for "identifier"; write appID instead of appId.